ipa-core/src/helpers/gateway/mod.rs

```diff
@@ -106,6 +106,26 @@ pub struct GatewayConfig {
     pub progress_check_interval: std::time::Duration,
 }
 
+impl ShardConfiguration for Gateway {
+    fn shard_id(&self) -> ShardIndex {
+        ShardConfiguration::shard_id(&self)
+    }
+
+    fn shard_count(&self) -> ShardIndex {
+        ShardConfiguration::shard_count(&self)
+    }
+}
+
+impl ShardConfiguration for &Gateway {
+    fn shard_id(&self) -> ShardIndex {
+        self.transports.shard.identity()
+    }
+
+    fn shard_count(&self) -> ShardIndex {
+        ShardIndex::from(self.transports.shard.peer_count() + 1)
```

Comment:
Can you add a comment saying we add 1 because these are my peers *excluding* me but shard count needs to count me or something like that?

Reply:
I think the code here is pretty self-explanatory, but I don't mind adding a comment